plugins/languages/cmake-tools.nix

```diff
@@ -0,0 +1,391 @@
+    plenary-nvim
+  ];
+
+  settingsOptions = {
+    cmake_command = helpers.defaultNullOpts.mkStr "cmake" ''
+      This is used to specify cmake command path.
+    '';
+
+    ctest_command = helpers.defaultNullOpts.mkStr "ctest" ''
+      This is used to specify ctest command path.
+    '';
+
+    cmake_regenerate_on_save = helpers.defaultNullOpts.mkBool true ''
+      Auto generate when save CMakeLists.txt.
+    '';
+
+    cmake_generate_options =
+      helpers.defaultNullOpts.mkAttributeSet ''
+        { "-DCMAKE_EXPORT_COMPILE_COMMANDS" = 1; }
+      ''
+      ''
+        This will be passed when invoke `CMakeGenerate`.
+      '';
+
+    cmake_build_options = helpers.defaultNullOpts.mkAttributeSet "{}" ''
+      This will be passed when invoke `CMakeBuild`.
+    '';
+
+    cmake_build_directory = helpers.defaultNullOpts.mkStr "out/\${variant:buildType}" ''
+      This is used to specify generate directory for cmake, allows macro expansion, relative to vim.loop.cwd().
+    '';
+
+    cmake_soft_link_compile_commands = helpers.defaultNullOpts.mkBool true ''
+      This will automatically make a soft link from compile commands file to project root dir.
+    '';
+
+    cmake_compile_commands_from_lsp = helpers.defaultNullOpts.mkBool false ''
+      This will automatically set compile commands file location using lsp, to use it, please set `cmake_soft_link_compile_commands` to false.
+    '';
+
+    cmake_kits_path = helpers.defaultNullOpts.mkStr null ''
+      This is used to specify global cmake kits path, see CMakeKits for detailed usage.
+    '';
+
+    cmake_variants_message =
+      helpers.mkCompositeOption ''
+        Settings for cmake variants messages.
+      ''
+      {
+        short =
+          helpers.defaultNullOpts.mkAttributeSet ''
+            { show = true; }
+          ''
+          ''
+            Whether to show short message.
+          '';
+
+        long =
+          helpers.defaultNullOpts.mkAttributeSet ''
+            { show = true; max_length = 40; }
+          ''
+          ''
+            Whether to show long message.
+          '';
+      };
+
+    cmake_dap_configuration =
+      helpers.defaultNullOpts.mkAttributeSet ''
+        {
+          name = "cpp";
+          type = "codelldb";
+          request = "launch";
+          stopOnEntry = false;
+          runInTerminal = true;
+          console = "integratedTerminal";
+        }
+      ''
+      ''
+        Dap configuration for cmake.
+      '';
+
+    cmake_executor =
+      helpers.mkCompositeOption ''
+        Executor to use.
+      ''
+      {
+        name = helpers.defaultNullOpts.mkStr "quickfix" ''
+          Name of the executor.
+        '';
+
+        opts = helpers.defaultNullOpts.mkAttributeSet "{}" ''
+          The options the executor will get, possible values depend on the executor type. See `default_opts` for possible values.
+        '';
+
+        default_opts =
+          helpers.defaultNullOpts.mkAttributeSet ''
```

Comment:
Would some of these `mkAttributeSet` options be better off as submodues? How do we normally handle this while sticking to [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md)?